lib/sisimai/bite/json/sendgrid.rb

```diff
@@ -0,0 +1,79 @@
+module Sisimai::Bite::JSON
+  # Sisimai::Bite::JSON::SendGrid parses a bounce object(JSON) which created by
+  # SendGrid. Methods in the module are called from only Sisimai::Message.
+  module SendGrid
+    # Imported from p5-Sisimail/lib/Sisimai/Bite/JSON/SendGrid.pm
+    class << self
+      require 'sisimai/bite/json'
+
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def description; return 'SendGrid(JSON): http://sendgrid.com/'; end
+
+      # @abstract Adapt SendGrid bounce object for Sisimai::Message format
+      # @param        [Hash] argvs  bounce object(JSON) retrieved from SendGrid API
+      # @return       [Hash, Nil]   Bounce data list and message/rfc822 part
+      #                             or Undef if it failed to parse or the
+      #                             arguments are missing
+      # @since v4.20.0
+      def adapt(argvs)
+        return nil unless argvs.is_a? Hash
+        return nil unless argvs.keys.size > 0
+        return nil unless argvs.key?('email')
+
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        rfc822head = {}     # (Hash) Check flags for headers in RFC822 part
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = dscontents[-1]
+
+        require 'sisimai/string'
+        require 'sisimai/address'
+
+        if Sisimai::RFC5322.is_emailaddress(argvs['email'])
+          #   {
+          #       "status": "4.0.0",
+          #       "created": "2011-09-16 22:02:19",
+          #       "reason": "Unable to resolve MX host sendgrid.ne",
+          #       "email": "esting@sendgrid.ne"
```

Comment:
__[rubocop]__
* Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression. [:ref](https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)

<!-- Comment by [SideCI](https://sideci.com) -->